### ISSUE TITLE
[Accessibility enhancement] ARIA attributes for clr-alert component

### ISSUE THREAD

**Issue by bashibozuk** [vmware/clarity#2582]
## Summary

I would like to request adding a aria attributes to the clr-alert component compliant with this recommendation https://www.w3.org/TR/wai-aria-practices-1.1/#alert
Important things are:
* role=alert, so assistive technologies recognize the element as alert
* aria-live=polite/assertive when the alert pops up assistive technologies must be able to read it or provide better solution for this case
##
* What is the change?
To enable assistive technologies to read clarity alerts when they appear
* Why should it go in Clarity?
Enhanced accessibility
* Does this change impact existing behaviors? If so how?
No
* If this change introduces a new behavior, is this behavior accessible?
N/A

## Use case
When user is at some page of a wizard he fills a input with incorrect value, then presses next.
Then server side validation is performed and a clr-alert is shown, but assistive technologies do not read it, and they should.

## Examples

N/A

## Workarounds

N/A

**Comment by mathisscott** [comment]
@bashibozuk
What screenreader are you using? We can not verify your assertion here and we would like to. Thanks.

**Comment by mathisscott** [comment]
@bashibozuk 
I've gone back over your assertion repeatedly with VoiceOver and cannot verify its veracity. If you are using just HTML/CSS alerts, then sure aria-live helps. But you can put aria-live on them yourself. If you are somehow finding success with using `clr-alert` components with a screenreader, it is likely an edge case and you are fortunate to be operating under that at this moment. 

Within a `clr-alert-item` you can put pretty much any content. So if you have a `span.alert-text` in there, you can try adding `aria-live` to those as a workaround. I have tried this myself and it doesn't register at all with the screenreader. But it appears your screenreader is picking them up somehow.

**Comment by bashibozuk** [comment]
@mathisscott i did not have time to cover the full  matrix of browser/os/screenreader , but I will do it on monday. My current findings are that `role=alert` with `aria-live=assertive` work with VoiceOver on Safary. I am pretty sure that it used to work on older verisons of Chrome and Firefox also, which makes me think that there is a regression regarding this in the latest versions of the browsers. Also that combination works wery well on Chrome on Windows with NVDA and JAWS.  I am also sure that it works on IE 11 with JAWS. Will share more details on monday.

**Comment by mathisscott** [comment]
@bashibozuk
I will test again with voiceover, focusing on Safari. I spent hours trying to verify this yesterday and could not. Specifically, I’m testing alerts that are dynamically inserted in the DOM using Angular. If you could provide a stackblitz with a working example, that would help a great deal.

**Comment by bashibozuk** [comment]
https://clarity-light-theme-v11-wevfgj.stackblitz.io/

  | JAWS | NVDA | Voice Over
-- | -- | -- | --
Firefox | Yes | Yes | No
IE | Yes | No | N/A
Safary | N/A | N/A | yes
Edge | No | No | N/A
Chrome | No | Yes | yes

**Comment by mathisscott** [comment]
Hi @bashibozuk 
I've verified that it works in Safari. But still cannot get this to work in Chrome. I can look into getting aria-live attributes in so we have something working at least in Safari. Not sure if this is something we can backport to 0.11. 0.12 and 0.13 are currently our supported versions.
